x-pack/plugins/infra/public/components/waffle/legend_controls.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSwitch,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import React, { SyntheticEvent, useState } from 'react';
+import styled from 'styled-components';
+import { InfraWaffleMapBounds } from '../../lib/lib';
+
+interface Props {
+  onChange: (options: { auto: boolean; bounds: InfraWaffleMapBounds }) => void;
+  bounds: InfraWaffleMapBounds;
+  dataBounds: InfraWaffleMapBounds;
+  autoBounds: boolean;
+  boundsOverride: InfraWaffleMapBounds;
+  intl: InjectedIntl;
+}
+
+export const LegendControls = injectI18n(
+  ({ intl, autoBounds, boundsOverride, onChange, dataBounds }: Props) => {
+    const [isPopoverOpen, setPopoverState] = useState(false);
+    const [draftAuto, setDraftAuto] = useState(autoBounds);
+    const [draftBounds, setDraftBounds] = useState(boundsOverride); // should come from bounds prop
```

Comment:
As the default state for the hook here is `boundsOverride` which originally comes from the initial state of the reducer (if not set in the URL) with `boundsOverride: { max: 1, min: 0 }`, it means the inputs have their default values set to `0` and `1` if auto bounds is `false`, however when auto bounds is toggled the inputs set their default values to the bounds of the data with `setDraftBounds(dataBounds);`. 

I think it would be nicer if the default input values always matched the bounds actually in use, regardless of whether they're auto or manual. Something like `const [draftBounds, setDraftBounds] = useState(autoBounds ? dataBounds : boundsOverride);`

![mount-state](https://user-images.githubusercontent.com/471693/53822428-9d9cef80-3f67-11e9-9b91-84c938026e63.gif)

Reply:
Good suggestion on initial `draftBounds` depending on if your starting out with `autoBounds === true`